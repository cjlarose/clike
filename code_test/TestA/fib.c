
void printint(int);
void printchar(char);

int fib(n)
  int n;
{
  if (n<=1) return n;

  return fib(n-1) + fib(n-2);
}

void main()
{

  int x;

  x = fib(0);
  printint(x);
  printchar(10);

  x = fib(1);
  printint(x);
  printchar(10);

  x = fib(2);
  printint(x);
  printchar(10);

  x = fib(3);
  printint(x);
  printchar(10);

  x = fib(4);
  printint(x);
  printchar(10);

  x = fib(5);
  printint(x);
  printchar(10);

  x = fib(6);
  printint(x);
  printchar(10);
}
