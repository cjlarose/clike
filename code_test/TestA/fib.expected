Loaded: /usr/local/share/spim/exceptions.s
0
1
1
2
3
5
8
